src/index.ts

```diff
@@ -165,6 +165,53 @@ const initialize = (apiKey: string, target: Target, options?: Options): Result =
 
   let evaluations: Record<string, Evaluation> = {}
 
+  const sendEvent = (evaluation: Evaluation) => {
+    logDebug('creating event for ', evaluation.identifier)
+
+    eventBus.emit(
+      Event.CHANGED,
+      hasProxy
+        ? new Proxy(evaluation, {
+          get(_flagInfo, property) {
+            if (_flagInfo.hasOwnProperty(property) && property === 'value') {
+              // only track metric when value is read
+              const featureIdentifier = _flagInfo.flag
+              const featureValue = evaluation.value
+              const entry = metrics.find(
+                _entry => _entry.featureIdentifier === featureIdentifier && _entry.featureValue === featureValue
+              )
+
+              if (entry) {
+                updateMetrics(entry)
+                entry.variationIdentifier = evaluations[featureIdentifier]?.identifier || ''
+              } else {
+                metrics.push({
+                  featureIdentifier,
+                  featureValue: String(featureValue),
+                  variationIdentifier: evaluations[featureIdentifier].identifier || '',
+                  count: metricsCollectorEnabled ? 1 : 0,
+                  lastAccessed: Date.now()
+                })
+              }
+              logDebug(
+                'Metrics event: Flag',
+                property,
+                'has been read with value via stream update',
+                featureValue
+              )
+            }
+
+            return property === 'value' ? convertValue(evaluation) : evaluation[property]
+          }
+        })
+        : {
+          deleted: evaluation.deleted,
+          flag: evaluation.flag,
+          value: convertValue(evaluation)
+        }
```

Comment:
Might just be personal preference, but this looks way too big to use a ternary IMHO. Consider using a more readable `if`/`else` instead

Reply:
cool I'd refactored that out of an existing block and lifted as is, but make sense to change that for readability now